arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+        'sin': 'جا',
+        'cos': 'جتا',
+        'tan': 'ظا',
+        'cot': 'ظتا',
+        'sec': 'قا',
+        'csc': 'قتا',
+        'log': 'لو'
+      },
+      // Limits
+      operatorsMap: {
+        'lim': 'نهــا'
+      }
+    });
+
+  MathJax.Extension.Arabic.config = MathJax.Hub.CombineConfig('Arabic', {
+    dict: {
+      // A macros to force English zero in both languages
+      "Zero": ["zero", Text('0', 'صفر')],  // Better localized Zero
+      "Radius": ["radius", Text('r', 'نق')],  // Circle radius
+      "Area": ["Area", Text('A', 'م')]  // Area of circles and other stuff
+    }
+  });
+});
+
+MathJax.Extension.Arabic = {
+  version: '1.0.0',
+  config: MathJax.Hub.CombineConfig("Arabic", {
+    dict: {},
+    isArabicPage: (document.documentElement.lang === 'ar'),
+    identifiersMap: {},
+    numbersMap: {
+      '0': '٠',
+      '1': '١',
+      '2': '٢',
+      '3': '٣',
+      '4': '٤',
+      '5': '٥',
+      '6': '٦',
+      '7': '٧',
+      '8': '٨',
+      '9': '٩'
+    },
+    operatorsMap: {
+      // English to Arabic punctuations
+      ',': '،',
+      ';': '؛'
+    }
+  }),
+  arabicLanguageRegExp: /([\u0600-\u06FF]+)/g,
+  TeX: function (english, arabic) {
+    // Creates a translated TeX macro.
+
+    return function (name) {
+      var TEX = MathJax.InputJax.TeX;
+
+      var tex;
+      if ('ar' === this.stack.env.lang) {
+        tex = arabic;
+      } else {
+        tex = english;
+      }
+
+      this.Push(TEX.Parse(tex).mml());
+    };
+  },
+  Text: function (english, arabicText) {
+    // Creates a translated TeX macro, with an Arabic plain text.
+
+    return MathJax.Extension.Arabic.TeX(english, '\\fliph{\\text{' + arabicText + '}}');
+  },
+  Symbols: function (english, arabicSymbols) {
+    // Creates a translated TeX macro that converts Arabic symbols into text nodes,
+    // and treats everything else as normal TeX.
+    var arabic = arabicSymbols.replace(
+      MathJax.Extension.Arabic.arabicLanguageRegExp,
+      '\\fliph{\\text{$1}}'
+    );
+
+    return MathJax.Extension.Arabic.TeX(english, arabic);
+  }
+};
+
+MathJax.Hub.Startup.signal.Post('Arabic TeX Startup');
+
+MathJax.Hub.Register.StartupHook('TeX Jax Ready', function () {
+  MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+    var TEX = MathJax.InputJax.TeX;
+    var Arabic = MathJax.Extension.Arabic;
+    var texParsePush = TEX.Parse.prototype.Push;
+    var texParseMMLToken = TEX.Parse.prototype.mmlToken;
+    var texParseAlignedArray = TEX.Parse.prototype.AlignedArray;
+    var dict = MathJax.Hub.config.Arabic.dict;
+
+    var englishNumbersRegExp = /[0-9]/g;
+
+    var escapeRegExp = (function () {
+      var reRegExpChar = /[\\^$.*+?()[\]{}|]/g;
+      var reHasRegExpChar = new RegExp(reRegExpChar.source);
+
+      return function (string) {
+        return (string && reHasRegExpChar.test(string))
+          ? string.replace(reRegExpChar, '\\$&')
+          : string;
+      };
```

Comment:
Is checking for a special character and then replacing if one is found really any faster than just replacing in the first place?  (I haven't done any testing to see, but I suspect it's not.)  Why not just

    escapeRegExp = function (string) {return string.replace(reRegExpChar,'\\$&');

in place of lines 243 to 252?